File: drestaurant-libs/drestaurant-customer/src/main/java/com/drestaurant/customer/domain/CustomerCommandHandler.java
package com.drestaurant.customer.domain;

import static org.axonframework.eventhandling.GenericEventMessage.asEventMessage;

import org.axonframework.commandhandling.CommandHandler;
import org.axonframework.commandhandling.model.Aggregate;
import org.axonframework.commandhandling.model.AggregateNotFoundException;
import org.axonframework.commandhandling.model.Repository;
import org.axonframework.eventhandling.EventBus;

public class CustomerCommandHandler {

	private Repository<Customer> repository;
	private EventBus eventBus;

	public CustomerCommandHandler(Repository<Customer> repository, EventBus eventBus) {
		this.repository = repository;
		this.eventBus = eventBus;
	}

	@CommandHandler
	public void handle(ValidateOrderByCustomerCommand command) {
		try {
			Aggregate<Customer> customerAggregate = repository.load(command.getCustomerId());
			customerAggregate.execute(customer -> customer.validateOrder(command.getOrderId(), command.getOrderTotal(), command.getAuditEntry()));
		} catch (AggregateNotFoundException exception) {
			eventBus.publish(asEventMessage(new CustomerNotFoundForOrderEvent(command.getCustomerId(), command.getOrderId(), command.getOrderTotal(), command.getAuditEntry())));
		}

	}

}

File: drestaurant-libs/drestaurant-customer/src/main/kotlin/com/drestaurant/customer/domain/CustomerCommandHandler.kt
package com.drestaurant.customer.domain

import org.axonframework.commandhandling.CommandHandler
import org.axonframework.eventhandling.EventBus
import org.axonframework.eventhandling.GenericEventMessage.asEventMessage
import org.axonframework.modelling.command.AggregateNotFoundException
import org.axonframework.modelling.command.Repository

internal open class CustomerCommandHandler(private val repository: Repository<Customer>, private val eventBus: EventBus) {

    @CommandHandler
    fun handle(command: ValidateOrderByCustomerInternalCommand) = try {
        repository.load(command.customerId.identifier).execute { it.validateOrder(command.targetAggregateIdentifier, command.orderTotal, command.auditEntry) }
    } catch (exception: AggregateNotFoundException) {
        eventBus.publish(asEventMessage<Any>(CustomerNotFoundForOrderInternalEvent(command.customerId, command.targetAggregateIdentifier, command.orderTotal, command.auditEntry)))
    }
}
